packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3256 @@
+  }
+
+  @override
+  String? get debugCurrentItem {
+    String? result;
+    assert(() {
+      if (openMenu != null) {
+        result = openMenus.map<String>((_MenuNode node) => node.toStringShort()).join(' > ');
+      }
+      return true;
+    }());
+    return result;
+  }
+
+  @override
+  String? get debugFocusedItem {
+    String? result;
+    assert(() {
+      if (primaryFocus?.context != null) {
+        result = _focusNodes[primaryFocus]?.toStringShort();
+      }
+      return true;
+    }());
+    return result;
+  }
+}
+
+// The InheritedWidget marker for _MenuBarController, used to find the nearest
+// ancestor _MenuBarController.
+class _MenuBarControllerMarker extends InheritedWidget {
+  const _MenuBarControllerMarker({
+    required this.controller,
+    required super.child,
+  });
+
+  final _MenuBarController controller;
+
+  @override
+  bool updateShouldNotify(covariant _MenuBarControllerMarker oldWidget) {
+    return controller != oldWidget.controller;
+  }
+}
+
+// A widget used as the main widget for the overlay entry in the
+// _MenuBarController. Since the overlay is a Stack, this widget produces a
+// Positioned widget that fills the overlay, containing its own Stack to arrange
+// the menus with. Positioning of the top level submenus is relative to the
+// position of the menu buttons.
+class _MenuStack extends StatelessWidget {
+  const _MenuStack(this.controller);
+
+  final _MenuBarController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned.fill(
+      child: FocusScope(
+        node: controller.overlayScope,
+        child: Actions(
+          actions: <Type, Action<Intent>>{
+            NextFocusIntent: _MenuNextFocusAction(controller: controller),
+            PreviousFocusIntent: _MenuPreviousFocusAction(controller: controller),
+            DirectionalFocusIntent: _MenuDirectionalFocusAction(
+              controller: controller,
+            ),
+            DismissIntent: _MenuDismissAction(controller: controller),
+            VoidCallbackIntent: VoidCallbackAction(),
+          },
+          child: Shortcuts(
+            // These are here to make sure that these override any shortcut
+            // bindings from the menu items when a menu is open. If someone
+            // wants to bind an arrow or tab to a menu item, it would otherwise
+            // override the default traversal keys. We want their shortcuts to
+            // apply everywhere but override these in the menu itself, since
+            // there we have to be able to traverse the menus.
+            shortcuts: _kMenuTraversalShortcuts,
+            child: _MenuBarControllerMarker(
+              controller: controller,
+              child: Stack(
+                children: <Widget>[
+                  if (controller.openMenus.isNotEmpty)
+                    ...controller.openMenus.where((_MenuNode node) => node.menuBuilder != null).map<Widget>(
```

Comment:
Wondering: what about items that don't have a menuBuilder? Would that be a bug here?

Reply:
No, not a bug. That's why the `where` filter is here.  Only items that have a submenu will have a `menuBuilder`, and only the submenus need to have an entry in this stack.